### PR TITLE
Add inverse distance weighting option to superob function

### PR DESCRIPTION
**Pull request by ypwang19**
**Description:**

1. The superob function now incorporates the capability for inverse distance weighted averaging. If optional parameters are provided when calling the function, it conducts weighted averaging; otherwise, it defaults to simple averaging.

3. The VIIRS converter has been enhanced to utilize this new functionality. Weighted averaging is only used in the computation of superobbed ObsValue and ObsError.

### CODE REVIEW

File: utils/obsproc/superob.h [hunk 1]
```diff
@@ -8,11 +8,47 @@
...
+        double longitude;
+    };
+    const double EARTH_RADIUS_KM = 6371.0;
+
+    // Function to convert degrees to radians
+    double toRadians(double degree) {
+        return degree * (M_PI / 180.0);
+    }
+
+    // Function to calculate the Haversine distance between two points
```
Comment (guillaumevernieres) [inline utils/obsproc/superob.h]: You should leverage functions from `oops` and/or `atlas` for this. 
Look in the `atlas` source code for `Earth::distance`.

File: utils/obsproc/superob.h [hunk 2]
```diff
@@ -8,11 +8,47 @@
 
 namespace gdasapp {
   namespace superobutils {
+    // Function to compute distance between two points
+    struct GeoPoint {
```
Comment (guillaumevernieres) [inline utils/obsproc/superob.h]: make use of `eckit::geometry::Point2` instead of declaring a new data structure.

File: utils/obsproc/superob.h [hunk 3]
```diff
@@ -8,11 +8,47 @@
...
+        double latitude;
+        double longitude;
+    };
+    const double EARTH_RADIUS_KM = 6371.0;
```
Comment (guillaumevernieres) [inline utils/obsproc/superob.h]: use `atlas::Earth` ... or something ... But the earht geometry is already defined in atlas, so no need to redefine the earht radius again.

**Review by guillaumevernieres — COMMENTED** [review]
Looks good, but it might be a good idea to start poking at what's available in `atlas`. You shouldn't have to redefine things like the earth radius. Distance on the sphere is already an implemented function somewhere in atlas.

File: utils/obsproc/superob.h (hunk 3 above)
Comment (ypwang19) [reply utils/obsproc/superob.h]: Thank you for the suggestion, theses are very useful information. I will check functions available in altas and eckit.

File: utils/obsproc/superob.h (hunk 1 above)
Comment (ypwang19) [reply utils/obsproc/superob.h]: Distance function from atlas is used in the updated code.

**Review by guillaumevernieres — APPROVED** [review]
Nice!

**Review by CoryMartin-NOAA — APPROVED** [review]
Looks great, thanks!

**Pull request merged**